backend/core/src/listings/listings.service.ts

```diff
@@ -137,6 +162,66 @@ export class ListingsService {
     if (!result) {
       throw new NotFoundException()
     }
+    if (lang !== Language.en) {
+      // Get key-value pairs from listing to be translated
+      const translations = this.getTranslations(result)
+
+      const translatedValues = await this.translateService().translate(translations.values, {
+        from: Language.en,
+        to: lang,
+      })
+      // Attach translated values to the listing
+      translations.keys.forEach((key, index) => {
+        this.setValue(result, key, translatedValues[0][index])
+      })
+    }
+
+    return result
+  }
+
+  private translateService = () => {
+    return new Translate({
+      credentials: {
+        private_key: process.env.GOOGLE_API_KEY.replace(/\\n/gm, "\n"),
+        client_email: process.env.GOOGLE_API_EMAIL,
+      },
+      projectId: process.env.GOOGLE_API_ID,
+    })
+  }
+  // Sets value to the object by string path. eg. "property.accessibility" or "preferences.0.title"
+  private setValue = (object, path, value) =>
+    path
+      .split(".")
+      .reduce((o, p, i) => (o[p] = path.split(".").length === ++i ? value : o[p] || {}), object)
```

Comment:
Can we use more verbose variable names here? Hard to parse what it's doing… 😅

Reply:
`o` and `p` corresponded with `object` and `path` but I've renamed them to be more direct as requested.